addons/project/models/project.py

```diff
@@ -1139,7 +1139,7 @@ def write(self, vals):
                     recurrence = self.env['project.task.recurrence'].create(rec_values)
                     task.recurrence_id = recurrence.id
 
-        if 'recurring_task' in vals and not vals.get('recurring_task'):
+        if 'recurring_task' in vals and not vals.get('recurring_task') and not self.env.context.get('unlinking_recurrence'):
             self.recurrence_id.unlink()
```

Comment:
Alternatively, we could maybe do this to avoid using a context key. 🤷🏼‍♂️ 
```suggestion
        if vals.get('recurring_task', True) and self.recurrence_id:
            tasks_in_recurrence = self.recurrence_id.task_ids
            self.recurrence_id.unlink()
            tasks_in_recurrence.write({'recurring_task': False})
```
What do you think? 🙂

Reply:
Well it doesn't pass the test I've written for the use-case, so I guess it's not applicable. ;(
(I would also prefer a solution without playing with the context, but I don't see a better way)

Reply:
Oh it is because I did a mistake in my previous review. 🙈 
```suggestion
        if not vals.get('recurring_task', True) and self.recurrence_id:
            tasks_in_recurrence = self.recurrence_id.task_ids
            self.recurrence_id.unlink()
            tasks_in_recurrence.write({'recurring_task': False})
```

Reply:
Ow crap, I didn't see that.
Your proposition works great, and it's way cleaner.
I've pushed your suggestion :)